### DISCUSSION TITLE
Should we remove faction camp system?

### DISCUSSION THREAD

**Discussion by scarf005** [cataclysmbn/Cataclysm-BN#3866] [Polls]
## Summary

We're opening a vote to determine whether to remove faction camp system from game.

## Rationale

1. According to some users, faction camp systems are rarely used, and many features are unfinished.
2. Faction camp system complicates NPC code, which makes improving NPC AI very hard
3. The role of faction camp overlaps with zone system, which is more stable, adopted widely, and proven to be very useful.
4. Removing this system _may_ help us develop faster, design more robust system. For example, we could divert development effort spent fixing faction camp bugs to improving zone system.

> [!Warning] 
> Please note that we cannot guarantee replacement system for faction camps due to time and effort constraints.

## What would happen if it gets accepted

if majority agrees on its removal, and the changes are made, end users will face following changes:

- existing faction camps will not work anymore.
- you will not be able to create new faction camps.
- it might not be possible to load old/new saves.

the actual PR isn't made yet (due to its scale), but when it gets open we'll link it here and update the progress.

## When the vote lasts

The poll will last from 2023-12-10/2023-12-17, [anywhere on earth](https://time.is/Anywhere_on_Earth).

## Extra

> [!Note]
> It would be really helpful for us if you could share the reason you agree/disagree.

**Comment by Vollch** [comment]
No. Rationale: 
It might be unfinished, but conceptually it's an interesting mechanic which i'd like to see playable one day.
And porting updates for existing camps from DDA will take much less time and efforts, than rewriting it from scratch. Especially if no one is even planning to rewrite whole that mechanic.
But only as long as we have *where* to port it. 

And "complicates NPC code" is way too broad and vague. Everything complicates code. And it most case those complications boils down to extra `if` branch. Do you have *real* examples where removing camps will be beneficial?

**Reply by ghost** [reply]
Can you show us their improvements to camps? As an end user and not a contributor, I'd mainly like to see it have a better UI and easier to get into.

**Reply by Vollch** [reply]
There's a lot of them, you can skim through the list [there](https://github.com/CleverRaven/Cataclysm-DDA/pulls?q=is%3Apr+is%3Aclosed+label%3A%22Player+Faction+Base+%2F+Camp%22) or [there](https://github.com/CleverRaven/Cataclysm-DDA/pulls?page=2&q=is%3Apr+is%3Aclosed+basecamp).

**Comment by DrPariah** [comment]
I'm on the 'Keep' side here. Zone System is nice, and it's useful, but Faction Camps provide more than the Zones on their own. They provide an actual use for NPCs, a few QOL improvements (scavenging food, etc.) if you choose to use them, and perhaps most importantly a sense of direction and goal to work on in a game that's very much make-your-own-fun bordering on emergent storytelling. Rycon's last Innawood series on YouTube would have been _far_ less interesting without Faction Camps to use, and while he plays DDA and not BN the point still stands.

I could readily see some potential benefits from removing it, but not as it's outlined here. This is too vague, based on 'maybes', 'could', and 'possibly' without a plan or roadmap or direction. And as helpfully pointed out under the Warning, nothing may actually be done after removal, so a chunk of functionality in the game could be yanked without any gain.

**Comment by borsek** [comment]
IMHO faction camps are useless. They invoke a completely new interface, work on different mechanics, and aren't really fleshed-out or enjoyable as is. I tried it in regular play, made a camp, built buildings, processed resources, etc., and beyond roleplaying purposes, it was useless.

If we moved crafting from camps directly to NPCs with zones - direct a NPC to dry all meat in zone A, drop it in zone B, for example, camps would be redundant.

I'm a bit on the fence, since I'm sure some people use them, but if I had to choose, I'd choose scrap them, personally. Due to the bugs, convoluted interface, and general uselessness of the NPCs I don't use them in playthroughs anyway, so faction camps also go unused. Their only benefit is infinite meat through trapping, but respawning wildlife, or wildlife hordes would replace that function anyway.

So, remove them, but only when we have wildlife respawns, and some form of directing NPCs to perform crafts using zones.

**Comment by DeinonychusDiscord** [comment]
As long as you can still issue orders via zone system [disassemble car, deconstruct furniture, do construction/deconstruction, etc] then there's nothing of value lost by getting rid of the actual camp system itself. To my knowledge the two are kind of intertwined [as far as dialogue interfacing goes.]

Faction camp system is the furthest thing from dynamic. For example, you cannot establish expansion tiles on terrain that isn't FIELD terrain. Fields often have trees in them and boulders, both of which would inhibit actual building. ROAD is an equally empty terrain type that often has wrecks or intact vehicles in them, also inhibiting actual buildings. Only one of these is actually considered valid to turn into an evolving structure foundation though.

As another example, you *cannot* assign NPCs to make most things until you have reached the requisite upgrade level of your given expansion buildings. This means you must waste in-game days gathering materials, staging materials, and then applying those materials to slowly fill in a forge, or a tannery, or a canteen, and slowly gain 3-5 new recipes each time that may or may not even be relevant to you. You want to make a fancy sword? Sorry, you're not upgraded enough... but here's some spikes and nails instead! You want to make some cookies? Sorry, you can't do that yet, but here's some wasteland sausage and cooked offal instead!

It's also horrifically impractical from a resource perspective. Why would you create a building from raw materials when you can simply take over an existing empty structure in a freshly cleared city? Or two? Or three? Or an entire block? That would make far more sense as an option, but it isn't even a possibility. No, you literally only can build entirely new structures from three truckloads of heavy materials. Those are materials much better used elsewhere.

I know other people are in favor of keeping the system. 

I think they would be satisfied by faction camps being replaced into part of the zone system - where instead of you building something from nothing, you have a simple check system for core functionals and if that passes, you can create a zone in the structure that passes. Check does HOUSE_001 at -007, 258 have COOKING functionals? Oven, heat source? Yes? Then assign COOK crafting zone where you like and unlock access to all cooking recipes the NPC can do.  Same thing with forging. Anvil? Forge? Forge fuel? Yes? Then assign FORGE crafting and unlock access to all fabrication recipes the NPC can do. 

Right now it's just way too resource intensive, tedious, and unrewarding to actually *build a camp as implemented.*

**Comment by MitztheKat** [comment]
As long as new systems will be put in place, I'm fully for removing fatty systems that slow down dev time.

**Comment by CounterDag** [comment]
I just want the ability to make a little town. Honestly, expanding the Zone system to incorporate the current benefits of faction camps would be better. Being able to set up workshop zones for specific crafts, living quarter zones, Patrol zones... You could allow for assigning zones to singular NPCs or groups of NPCs and we could have some discount Dwarf Fortress comfyness.

**Comment by chaosvolt** [comment]
I am smol and neutral on this, because I know that any solution that fixes the problem eventually is gonna take effort. Either rewriting and fleshing out the JSON for faction camps (which so far looks to be somehow even more annoying to work with than mission and dialogue JSON), hardcoding in new zones to replace old faction camp functions, or else adding support for JSONizing new types of zones in a way that actually makes it easier to implement. :D

**Comment by JBloodthorn** [comment]
It was started and mostly dropped before other major systems like the electrical grid were added. I think it should be redone with those systems in mind, if it offers value to the game.

**Comment by sayke** [comment]
I use the faction camp system extensively

**Comment by Mimmimmi** [comment]
I'm not contributer, but I liked camp system .. 
Without camp, NPC will be quiet useless in my play. I need my NPC slaves for wood & meat..

**Comment by DarianLStephens** [comment]
If faction camps get removed, then I'd like to see some of the main benefits stay.
Those being, a marked place to tell NPCs to head to (Including over radio), the ability to send them on missions which they will autonomously do without needing to be monitored (Because reality bubble), and 'continuous' tasks.

Zones are fine right now, and it would be really cool to granularly design a base with them (Construction zones and whatnot), but having to repeatedly tell your followers to do this task instead of them doing it on their own when they're able, or telling them to continue when something interrupts it early, gets grating, especially with the tedious number/letter system of menu navigation, and the oft poorly-labelled dialogue entries to even do them in the first place.

Plus, it just seems nice to have NPCs going from place to place doing stuff. The ability to place prefab structures is also great for quickly getting something up without placing every tile, and for the less building-oriented among us.
The radio tower could probably be one of those prefab structures, if a system involving it were created.

In fact, that could be good in general. Quick prefab buildings you can place, or even an entirely new zone system dedicated to construction so it doesn't get muddled in with every other zone and is easier to navigate. Being able to set a 'zone' to 'build this thing from the build menu' could be really interesting, like a 'building mode' in the game. Would certainly make building a lot nicer, and you could plan stuff out pretty nicely.
Since zones are persistent, it would also let you more easily set up long-term goals, because you can go back to the zone and look at your plans, and slowly build it out.

**Comment by mythosmod** [comment]
While I have experimented with faction camps and like the idea of them, the reality I have often found is that they are inflexible and resource hungry in a way that doesn’t suit my normal play style at all - I would be in favour of expanding the zone system further instead as that has the flexibility that the camps seem to lack and could incorporate similar mechanics (npc gathering\guarding\building\crafting) while supporting out of reality bubble time passing. For example, it might be possible to have something like a “construct wall”, “construct floor” or “construct roof” zone that npcs assigned to building treat as if there was a construction there and work on it. Just spitballing though, others may get much more mileage out of the existing systems than I do.

**Reply by DarianLStephens** [reply]
Maybe there could also be 'map zones', where you lay out plans for entire overmap tiles? That could be a substitute for things like the wood harvesting mission which actually depletes trees slowly, hunt animals in a general area, try to scavenge resources from a town with risks, allow you to potentially build roads or larger walls without planning the entire thing out piece by piece directly.

With things like overmap construction in particular, that could be a good way to deal with the wandering horde problem where they teleport through defences.
I recall plans ages ago to prevent hordes from moving through overmap tiles like that, maybe through detecting when walls cross from end to end of a tile and marking it as 'walled' on a particular axis. This would be a simple way to implement that without scanning for walls and doing it dynamically.

**Reply by ghost** [reply]
Yeah, I love the idea to do overmap changes. In practice this is going to be incredibly difficult to execute with our current codebase.

**Comment by Ignaramico** [comment]
I kinda like the faction camp for some benefits that provide in recipe making (like actions not being stopped because you sleep) and kind of giving some purpose to all the construction loot (but only if you used the pre defined options for the faction like the evac shelter) but aside from that there is no real use for it, even if you could make something useful on its options, you would need to go back and forth, limiting your exploring (that one of the reasons most of the times i made a simple one near a evac shelter since i would return because of the missions) and still with zone you can construct/deconstruct with blueprints

**Reply by ghost** [reply]
Teleportation is the only real way to fix static basing and there's a lua-based mod for it, and a few mods that add teleporter items/furnitures or spells based on the spell system. That's not an issue with faction camp but any static basing sadly, returning to a location takes too long.

**Comment by KheirFerrum** [comment]
My vote is to remove the faction camp system, but allow designation of home base zones without the rigid blueprint requirements of faction camps themselves. One of the biggest issues with faction camps is how rigid and unextensible they are. A more modular system that for instance looks for the tools/beds/appliances etc within the designated overmap tile would go a long way toward achieving what the faction camp system was meant to do without the baggage and work arounds needed to make the current faction camp system good.

**Comment by ghost** [comment]
One of the unspoken huge problems with faction camps is that because I added a new soldering iron and welder via modding, it broke dozens of camp blueprint recipes which throws errors. I don't know if that means you can no longer build the faction camp blueprints if you run my mod, but it's so inflexible that it errors out when you touch crafting standards at all.
